stdlib/hashtbl.ml

```diff
@@ -214,19 +215,15 @@ let fold f h init =
         accu
     | Cons{key; data; next} ->
         do_bucket next (f key data accu) in
-  let old_trav = ongoing_traversal h in
-  if not old_trav then flip_ongoing_traversal h;
-  try
+  protect_traversal h (fun () ->
     let d = h.data in
     let accu = ref init in
     for i = 0 to Array.length d - 1 do
       accu := do_bucket d.(i) !accu
     done;
     if not old_trav then flip_ongoing_traversal h;
```

Comment:
keeping this line is probably a bug

Reply:
(this is rather an issue in commit splitting, the present comment does not build but further commits remove the line as expected)